executor/delete.go

```diff
@@ -165,10 +171,16 @@ func (e *DeleteExec) removeRowsInTblRowMap(tblRowMap tableRowMapType) error {
 }
 
 func (e *DeleteExec) removeRow(ctx sessionctx.Context, t table.Table, h int64, data []types.Datum) error {
-	err := t.RemoveRecord(ctx, h, data)
+	// err is always nil if `active` is false.
+	txnState, err := e.ctx.Txn(false)
+	if err != nil {
```

Comment:
It's better to return error even 

> err is always nil if `active` is false.

Because you can't be sure one day this assertion doesn't hold any more.